perf/src/sigverify.rs

```diff
@@ -237,17 +239,88 @@ fn do_get_packet_offsets(
     ))
 }
 
-fn get_packet_offsets(packet: &Packet, current_offset: usize) -> PacketOffsets {
+fn get_packet_offsets(packet: &mut Packet, current_offset: usize) -> PacketOffsets {
     let unsanitized_packet_offsets = do_get_packet_offsets(packet, current_offset);
     if let Ok(offsets) = unsanitized_packet_offsets {
+        if let Some(msg_start_offset) = (offsets.msg_start as usize).checked_sub(current_offset) {
+            check_for_simple_vote_transaction(packet, msg_start_offset).ok();
+        }
         offsets
     } else {
         // force sigverify to fail by returning zeros
         PacketOffsets::new(0, 0, 0, 0)
     }
 }
 
-pub fn generate_offsets(batches: &[Packets], recycler: &Recycler<TxOffset>) -> TxOffsets {
+fn check_for_simple_vote_transaction(
+    packet: &mut Packet,
+    msg_start_offset: usize,
+) -> Result<(), PacketError> {
+    let message_account_keys_len_offset = msg_start_offset
+        .checked_add(MESSAGE_HEADER_LENGTH)
+        .ok_or(PacketError::InvalidLen)?;
+
+    let (pubkey_len, pubkey_len_size) =
+        decode_shortu16_len(&packet.data[message_account_keys_len_offset..])
+            .map_err(|_| PacketError::InvalidLen)?;
+
+    let pubkey_start = message_account_keys_len_offset
+        .checked_add(pubkey_len_size)
+        .ok_or(PacketError::InvalidLen)?;
```

Comment:
Can you mirror @sakridge's changes [here](https://github.com/solana-labs/solana/pull/20007/files#diff-c4686ade7f5e09568084fc8e5ff4df62961724ef52932e93a759e46d609b0d2fR49) which add a `pubkey_len` field to `PacketOffsets` and use that and the `pubkey_start` offset to replace this code.

Reply:
done